src/Files.App/Views/NavigationInteractionTracker.cs

```diff
@@ -0,0 +1,205 @@
+			forwardResistanceValue.SetReferenceParameter("tracker", _tracker);
+			forwardResistance.Condition = forwardResistanceCondition;
+			forwardResistance.Value = forwardResistanceValue;
+
+			List<CompositionConditionalValue> conditionalValues = new() { backResistance, forwardResistance };
+			_source.ConfigureDeltaPositionXModifiers(conditionalValues);
+
+			var backAnim = compositor.CreateExpressionAnimation("props.CanGoBack ? (-clamp(tracker.Position.X, -96, 0) * 2) - 48 : this.CurrentValue - 48");
+			backAnim.SetReferenceParameter("tracker", _tracker);
+			backAnim.SetReferenceParameter("props", _props);
+			_backVisual.StartAnimation("Translation.X", backAnim);
+
+			var forwardAnim = compositor.CreateExpressionAnimation("props.CanGoForward ? (-clamp(tracker.Position.X, 0, 96) * 2) + 48 : this.CurrentValue + 48");
+			forwardAnim.SetReferenceParameter("tracker", _tracker);
+			forwardAnim.SetReferenceParameter("props", _props);
+			_forwardVisual.StartAnimation("Translation.X", forwardAnim);
+		}
+
+		private void PointerPressed(object sender, PointerRoutedEventArgs e)
+		{
+			if (e.Pointer.PointerDeviceType == PointerDeviceType.Touch)
+			{
+				_source.TryRedirectForManipulation(e.GetCurrentPoint(_rootElement));
+			}
+		}
+
+		private class InteractionTrackerOwner : IInteractionTrackerOwner
+		{
+			private NavigationInteractionTracker _parent;
+			private bool _shouldBounceBack;
+
+			public InteractionTrackerOwner(NavigationInteractionTracker parent)
+			{
+				_parent = parent;
+			}
+
+			public void CustomAnimationStateEntered(InteractionTracker sender, InteractionTrackerCustomAnimationStateEnteredArgs args)
+			{
+
+			}
+
+			public void IdleStateEntered(InteractionTracker sender, InteractionTrackerIdleStateEnteredArgs args)
+			{
+				if (!_shouldBounceBack) return;
+
+				var compositor = _parent._rootVisual.Compositor;
+				var springAnim = compositor.CreateSpringVector3Animation();
+				springAnim.FinalValue = new(0f);
+				springAnim.DampingRatio = 1f;
+				_parent._tracker.TryUpdatePositionWithAnimation(springAnim);
+				_shouldBounceBack = false;
+
+				if (Math.Abs(sender.Position.X) < 64)
+					return;
+
+				EventHandler<SwipeNavigationEventArgs>? navEvent = _parent.NavigationRequested;
+				if (navEvent is null)
+					return;
+
+				navEvent(_parent, sender.Position.X > 0 ? SwipeNavigationEventArgs.Forward : SwipeNavigationEventArgs.Back);
+			}
+
+			public void InertiaStateEntered(InteractionTracker sender, InteractionTrackerInertiaStateEnteredArgs args)
+			{
+
+			}
+
+			public void InteractingStateEntered(InteractionTracker sender, InteractionTrackerInteractingStateEnteredArgs args)
+			{
+				_shouldBounceBack = true;
+			}
+
+			public void RequestIgnored(InteractionTracker sender, InteractionTrackerRequestIgnoredArgs args)
+			{
+
+			}
+
+			public void ValuesChanged(InteractionTracker sender, InteractionTrackerValuesChangedArgs args)
+			{
+
+			}
```

Comment:
What are these methods needed for?

Reply:
They're only there to comply with IInteractionTrackerOwner

Reply:
You may group them (maybe at the end of the class) and add a comment to explain why they are there